File: webpack/mutations/UserMutations.js
import { commitMutation, graphql } from 'react-relay';
import environment from '../environment';

const mutation = graphql`
  mutation UserMutationsMutation(
    $input: CreateUserInput!
  ) {
    CreateUser(input: $input) {
      viewer {
        name
        ...RoomList_viewer
      }
    }
  }
`;

export function createUser(name) {
  const variables = {
    input: {
      name
    },
  };

  commitMutation(
    environment,
    {
      mutation,
      variables
    }
  );
}

File: app/graphql/mutations/user_mutations.rb
module Mutations::UserMutations
  Create = GraphQL::Relay::Mutation.define do
    name 'CreateUser'
    description 'Create user'

    input_field :name, !types.String
    return_field :viewer, Types::ViewerType

    resolve -> (obj, inputs, ctx) {
      user = User.create!(name: inputs[:name])
      ctx[:controller].login(user)

      {
        viewer: user
      }
    }
  end
end
